NBitcoin.Tests/NetworkAddressTests.cs

```diff
@@ -0,0 +1,560 @@
+using System.IO;
+using System.Linq;
+using System.Threading;
+using NBitcoin;
+using Xunit;
+using System;
+using NBitcoin.Protocol;
+using System.Net;
+using NBitcoin.DataEncoders;
+
+namespace NBitcoin.Tests
+{
+	public class NetworkAddressTests
+	{
+		[Fact]
+		[Trait("UnitTest", "UnitTest")]
+		public void CanParseSpecialAddresses()
+		{
+			var addr = new NetworkAddress();
+
+			// TORv2
+			Assert.True(addr.SetSpecial("6hzph5hv6337r6p2.onion"));
+			Assert.True(addr.IsTor);
+
+			Assert.True(addr.IsAddrV1Compatible);
+			Assert.Equal("6hzph5hv6337r6p2.onion", addr.ToAddressString());
+
+			// TORv3
+			var torv3_addr = "pg6mmjiyjmcrsslvykfwnntlaru7p5svn6y2ymmju6nubxndf4pscryd.onion";
```

Comment:
What if somebody passes an address which ends with `.ONION` instead of `.onion`? Is that correct or not? Should there be a test for it?

Reply:
No, it cannot be uppercase.
> Valid onion addresses contain 16 characters in a-z2-7 plus ".onion"

Reply:
see:  [rend-spec-v3](https://github.com/torproject/torspec/blob/master/rend-spec-v3.txt) 6. Encoding onion addresses [ONIONADDRESS]

Reply:
Yeah, I have seen that after posting that comment. Still the test should be in place to not allow ".ONION".